src/back/spv/writer.rs

```diff
@@ -724,9 +769,13 @@ impl Writer {
     fn parse_type_declaration(
         &mut self,
         arena: &crate::Arena<crate::Type>,
-        handle: crate::Handle<crate::Type>,
+        lookup_type: &LookupType<crate::Type>,
     ) -> (Instruction, Word) {
-        let ty = &arena[handle];
+        let ty = match lookup_type {
```

Comment:
please don't match on a reference, instead let's match the type `match *lookup_type`